src/components/messageArchive/messageArchive.tsx

```diff
@@ -0,0 +1,236 @@
+import './messageArchive.css'
+
+import Alert from '@mui/material/Alert'
+import Chip from '@mui/material/Chip'
+import Box from '@mui/system/Box'
+import React, { useEffect, useRef, useState } from 'react'
+
+import ElementRenderer from '../elementRenderer/elementRenderer'
+import Icon from '../icon/icon'
+import MessageCanvas, {
+  type MessageContainerProps,
+} from '../messageCanvas/messageCanvas'
+import type { ComponentMap, Message } from '../types'
+
+export interface MessageArchiveProps extends MessageContainerProps {
+  /** A component map contains message formats as keys and their corresponding React components as values. */
+  supportedElements: ComponentMap
+  /** Previous chat messages */
+  receivedMessages: Message[]
+  /** Text label for scroll down button. Default value is 'scroll down'. */
+  scrollDownLabel?: string
+  /** Info message to display at the top of the message archive */
+  infoMessage?: string
+}
+
+function getCombinedMessages(
```

Comment:
This looks like duplicate code from MessageSpace. We could avoid that by using Composition or Higher-Order Components.